ansible_runner/display_callback/module.py

```diff
@@ -167,6 +170,17 @@ def clear_task(self, local=False):
 
     def v2_playbook_on_start(self, playbook):
         self.set_playbook(playbook)
+
+        # Record installed Collections + Collections version info
+        cmd = "ansible-galaxy collection list --format json"
```

Comment:
@matburt @ryanpetrello I didn't know where else to call this command to record the data we need. On a scale from 1-10, how dumb is this? We need to do it somewhere inside of the callback system so that we execute it inside of the EE. Halp.

Reply:
![](http://img3.wikia.nocookie.net/__cb20150424014315/degrassi/images/5/5a/Cover_eyes_shining.gif)

Reply:
If this _is_ an ok place to handle this, but the issue is mostly with the subprocess, maybe we need to reconsider writing out this metadata at build time. This would cause us to lose insight into collections installed via project updates.

Reply:
Yea, the notion to me of spawning another process *inside* of this one is a little gross to me 😞

Reply:
oof yeah, I don't think we can do this.

@ryanpetrello and I discussed this brad in the context of some other tooling yesterday.... both embedding this information in the EE OR potentially calling it as a pre-run command before other tasks.

Reply:
I actually had another idea last night after thinking about this more... We'll probably just end up putting this in Runner's entrypoint. Unless one of you have a better idea @ryanpetrello @matburt

Reply:
Yea, I think runner's entry point is a valid option.